systemd-udevd/pkg.yaml

```diff
@@ -0,0 +1,114 @@
+        patch -p1 < /pkg/patches/0016-Handle-missing-gshadow.patch
+        patch -p1 < /pkg/patches/0017-missing_syscall.h-Define-MIPS-ABI-defines-for-musl.patch
+        patch -p1 < /pkg/patches/0019-Adjust-for-musl-headers.patch
+        patch -p1 < /pkg/patches/0022-sd-event-Make-malloc_trim-conditional-on-glibc.patch
+        patch -p1 < /pkg/patches/0023-shared-Do-not-use-malloc_info-on-musl.patch
+        patch -p1 < /pkg/patches/0024-avoid-missing-LOCK_EX-declaration.patch
+        patch -p1 < /pkg/patches/0025-include-signal.h-to-avoid-the-undeclared-error.patch
+        patch -p1 < /pkg/patches/0026-undef-stdin-for-references-using-stdin-as-a-struct-m.patch
+        patch -p1 < /pkg/patches/0027-adjust-header-inclusion-order-to-avoid-redeclaration.patch
+        patch -p1 < /pkg/patches/0028-build-path.c-avoid-boot-time-segfault-for-musl.patch
+
+        # Own patches
+        patch -p1 < /pkg/patches/0001-chore-musl-headers-compatibility-for-udev.patch
+
+        # for full build add this:
+        #-Dnss-mymachines=disabled -Dnss-myhostname=false -Dnss-resolve=disabled -Dnss-systemd=false
+        meson setup build \
+          --buildtype=release \
+          -Dmode=release \
+          -Dversion-tag="{{ .systemd_version }}-talos" \
+          -Dselinux=enabled \
+          -Dblkid=enabled \
+          -Dkmod=enabled \
+          -Dopenssl=disabled \
+          -Dgshadow=false \
+          -Dutmp=false \
+          -Duserdb=false \
+          -Ddbus=disabled \
+          -Dglib=disabled \
+          -Dlink-udev-shared=false \
+          -Dman=disabled \
+          -Dtests=false \
+          -Dwerror=false
+    build:
+      - |
+        # rules.d/64-btrfs.rules is disabled in our build, other are static
+        # Buildable rules from rules.d/meson.build
+        ninja -j $(nproc) -C build udevadm rules.d/50-udev-default.rules rules.d/71-seat.rules rules.d/90-vconsole.rules \
+          rules.d/60-persistent-storage.rules rules.d/70-uaccess.rules rules.d/73-seat-late.rules rules.d/99-systemd.rules \
+          ata_id cdrom_id dmi_memory_id fido_id iocost mtd_probe scsi_id v4l_id \
+          libudev.so.1.7.9 libudev.a src/libudev/libudev.pc src/udev/udev.pc
+        # Build all instead and copy necessary parts?
+        # ninja -j $(nproc) -C build
+    install:
+      - |
+        mkdir -p /rootfs/usr/bin
+        cp build/udevadm /rootfs/usr/bin/
+        mkdir -p /rootfs/sbin
+        ln -s /rootfs/usr/bin/udevadm /rootfs/sbin/udevadm
+        ln -s /rootfs/usr/bin/udevadm /rootfs/sbin/systemd-udevd
+
+        mkdir -p /rootfs/usr/lib/udev/rules.d/
+        cp build/rules.d/*.rules /rootfs/usr/lib/udev/rules.d/
+        cp rules.d/*.rules /rootfs/usr/lib/udev/rules.d/
+        rm -f /rootfs/usr/lib/udev/rules.d/{60-cdrom_id.rules,60-persistent-alsa.rules,60-persistent-v4l.rules,64-btrfs.rules,70-joystick.rules,70-mouse.rules,70-touchpad.rules,78-sound-card.rules,80-net-name-slot.rules}
```

Comment:
We might want to remove more rules from our package.

Left: systemd-udevd, right: eudev:
![image](https://github.com/user-attachments/assets/eaa45e6e-6521-4696-8dcd-4c9a7ff3882a)

Reply:
reviewing the following extra rules: `60-dmi-id.rules 60-infiniband.rules 60-persistent-media-controller.rules 60-persistent-storage-mtd.rules 70-power-switch.rules 70-uaccess.rules 71-seat.rules 73-seat-late.rules 80-net-setup-link.rules 82-net-auto-link-local.rules 90-iocost.rules 90-vconsole.rules 99-systemd.rules`

Reply:
`99-systemd.rules` might be useless for us as it just points which targets should be activated. `71-seat.rules` might actually cause issues as we don't have logind. `90-vconsole.rules` also refers to systemd binaries and parts we have no intention to include.

Reply:
Current file list: https://tpaste.us/1yxz